jobs/webcompat-kb/webcompat_kb/main.py

```diff
@@ -78,283 +100,543 @@ def get_urls_from_story(user_story_str):
 
-    url = f"{BUGZILLA_API}/bug"
-    params["include_fields"] = ",".join(fields)
+        match = re.search(r"\d+", field)
+        if match:
+            return int(match.group())
 
-    response = requests.get(url, params=params)
-    response.raise_for_status()
-    result = response.json()
+    return None
 
-    return result["bugs"]
 
+def parse_string_to_json(input_string):
+    if not input_string:
+        return ""
+
+    input_string = input_string.replace("\r\n", "\n")
+    lines = input_string.split("\n")
+
+    result_dict = {}
+
+    for line in lines:
+        if line:
+            key_value = line.split(":", 1)
+            if len(key_value) == 2:
+                key, value = key_value
+                if key in result_dict:
+                    if isinstance(result_dict[key], list):
+                        result_dict[key].append(value)
+                    else:
+                        result_dict[key] = [result_dict[key], value]
+                else:
+                    result_dict[key] = value
+    if not result_dict:
+        return ""
+
+    return result_dict
+
+
+class BugzillaToBigQuery:
+    def __init__(self, bq_project_id, bq_dataset_id, bugzilla_api_key):
+        self.client = bigquery.Client(project=bq_project_id)
+        self.bq_dataset_id = bq_dataset_id
+        self.bugzilla_api_key = bugzilla_api_key
+        self.bugs_fetch_completed = True
+        self.history_fetch_completed = True
+
+    def fetch_bugs(self, params=None):
+        if params is None:
+            params = {}
+
+        fields = [
+            "id",
+            "summary",
+            "status",
+            "resolution",
+            "product",
+            "component",
+            "see_also",
+            "depends_on",
+            "blocks",
+            "priority",
+            "severity",
+            "creation_time",
+            "keywords",
+            "url",
+            "cf_user_story",
+            "cf_last_resolved",
+            "last_change_time",
+            "whiteboard",
+        ]
+
+        headers = {}
+        if self.bugzilla_api_key:
+            headers = {"X-Bugzilla-API-Key": self.bugzilla_api_key}
+
+        url = f"{BUGZILLA_API}/bug"
+        params["include_fields"] = ",".join(fields)
+
+        try:
+            response = requests.get(url, params=params, headers=headers)
+            response.raise_for_status()
+            result = response.json()
+            return result["bugs"]
+        except Exception as e:
+            logging.error(f"Error: {e}")
+            self.bugs_fetch_completed = False
+            return []
+
+    def fetch_all_bugs(self):
+        fetched_bugs = {}
+
+        for category, filter_config in FILTER_CONFIG.items():
+            logging.info(f"Fetching {category.replace('_', ' ').title()} bugs")
+            fetched_bugs[category] = self.fetch_bugs(filter_config)
+
+        kb_bugs = []
+        kb_depends_on_ids = set()
+
+        for bug in fetched_bugs["wc"]:
```

Comment:
The script only fetches blocking bugs of KB bugs (`depends_on` field) and doesn't fetch dependencies of Site reports, so this shouldn't be a problem.

```python
        for bug in fetched_bugs["wc"]:
            if bug["component"] == "Knowledge Base":
                kb_bugs.append(bug)
                kb_depends_on_ids.update(bug["depends_on"])
     
        core_bugs = self.fetch_bugs({"id": ",".join(map(str, kb_depends_on_ids))})

        all_bugs = (
            fetched_bugs["wc"]
            + fetched_bugs["interventions"]
            + fetched_bugs["other"]
            + core_bugs
        )
```
It made me think about a scenario where there could be duplicates - a core bug has `webcompat:` keyword and it's also a blocking bug for a KB entry (so it will be in both `fetched_bugs["other"]` and `core_bugs`). We don't have these kind of bugs right now, but it could be possible in the future.

For example, https://bugzilla.mozilla.org/show_bug.cgi?id=1796300 or https://bugzilla.mozilla.org/show_bug.cgi?id=1884638 both are core bugs with `webcompat:needs-diagnosis` with no kb entry (yet). Though in this case, we should probably remove the keyword once a bug is moved to core (i.e. diagnosed or can't be diagnosed further), or once kb entry is created.

Either way, I could deduplicate `all_bugs` by bug number before saving to avoid importing duplicates like that or similar :)